src/dragonball/src/config_manager.rs

```diff
@@ -0,0 +1,723 @@
+// Copyright (C) 2020-2022 Alibaba Cloud. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::convert::TryInto;
+use std::io;
+use std::ops::{Index, IndexMut};
+use std::sync::Arc;
+
+use dbs_device::DeviceIo;
+use dbs_utils::rate_limiter::{RateLimiter, TokenBucket};
+use serde_derive::{Deserialize, Serialize};
+
+/// Trait for generic configuration information.
+pub trait ConfigItem {
+    /// Related errors.
+    type E;
+
+    /// Get the unique identifier of the configuration item.
+    fn id(&self) -> &str;
+
+    /// Check whether current configuration item conflicts with another one.
+    fn check_conflicts(&self, other: &Self) -> std::result::Result<(), Self::E>;
+}
+
+/// Struct to manage a group of configuration items.
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct ConfigInfos<T>
+where
+    T: ConfigItem + Clone,
+{
+    configs: Vec<T>,
+}
+
+impl<T> ConfigInfos<T>
+where
+    T: ConfigItem + Clone,
+{
+    /// Constructor
+    pub fn new() -> Self {
+        ConfigInfos {
+            configs: Vec::new(),
```

Comment:
```
pub struct ConfigInfos<T>
where
    T: ConfigItem + Clone,
{
    configs: Vec<T>,
}
```

Since `T` here doesn't have `Default trait`, we could not change this line to `ConfigInfos::default()`

Reply:
Yes, it works~~~I have changed this to `ConfigInfos::default()`